src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.WebSearch/Main.cs

```diff
@@ -0,0 +1,412 @@
+        {
+            return Properties.Resources.plugin_name;
+        }
+
+        public string GetTranslatedPluginDescription()
+        {
+            return Properties.Resources.plugin_description;
+        }
+
+        private void OnThemeChanged(Theme oldtheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+            UpdateBrowserIconPath(newTheme);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Design",
+            "CA1031:Do not catch general exception types",
+            Justification = "We want to keep the process alive but will log the exception")]
+        private void UpdateBrowserIconPath(Theme newTheme)
+        {
+            try
+            {
+                string progId = GetRegistryValue(
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice",
+                    "ProgId");
+
+                if (progId.StartsWith("Opera", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Opera user preferences file:
+                    string prefFile;
+
+                    if (progId.Contains("GX", StringComparison.OrdinalIgnoreCase))
+                    {
+                        _browserName = "Opera GX";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera GX Stable\\Preferences");
+                    }
+                    else
+                    {
+                        _browserName = "Opera";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera Stable\\Preferences");
+                    }
+
+                    string url = ExtraStringMethods.GetJSONStringValue(prefFile, new string[] { "default_search_provider_data", "template_url_data", "url" });
+
+                    if (url is null)
+                    {
+                        // "default_search_provider_data" doesn't exist in Preferences if the user hasn't searched for the first time,
+                        // therefore we set `url` to opera's default search engine.
+                        url = "https://www.google.com/search?client=opera&q={0}&sourceid=opera";
+                    }
+                    else
+                    {
+                        url = url
+                            .Replace("{searchTerms}", "{0}", StringComparison.Ordinal)
+                            .Replace("{inputEncoding}", "UTF-8", StringComparison.Ordinal)
+                            .Replace("{outputEncoding}", "UTF-8", StringComparison.Ordinal);
+
+                        // just in case there are other url parameters
+                        for (int i = ExtraStringMethods.GetStrNthIndex(url, '}', 2); i != -1; i = ExtraStringMethods.GetStrNthIndex(url, '}', 2))
```

Comment:
This feels breaky.  Why isn't this in a class?  I see magic index numbers that aren't clear what they mean

Reply:
this removes extra url parameters without a value (like `&foo={bar}`). not sure how to simplify this